parts/linux/cloud-init/artifacts/validate-kubelet-credentials.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -euo pipefail
+
+# this gives us logs_to_events and retry wrappers
+source /opt/azure/containers/provision_source.sh 
+
+KUBECONFIG_PATH="${KUBECONFIG_PATH:-/var/lib/kubelet/kubeconfig}"
+BOOTSTRAP_KUBECONFIG_PATH="${BOOTSTRAP_KUBECONFIG_PATH:-/var/lib/kubelet/bootstrap-kubeconfig}"
+
+VALIDATE_KUBELET_CREDENTIALS_MAX_RETRIES=${VALIDATE_KUBELET_CREDENTIALS_MAX_RETRIES:-10}
+VALIDATE_KUBELET_CREDENTIALS_RETRY_DELAY_SECONDS=${VALIDATE_KUBELET_CREDENTIALS_RETRY_DELAY_SECONDS:-3}
+VALIDATE_KUBELET_CREDENTIALS_RETRY_TIMEOUT_SECONDS=${VALIDATE_KUBELET_CREDENTIALS_RETRY_TIMEOUT_SECONDS:-5}
+
+function validateKubeconfig {
+    local kubeconfig_path=$1
+
+    if ! retrycmd_if_failure $VALIDATE_KUBELET_CREDENTIALS_MAX_RETRIES \
```

Comment:
is 30sec long enough ?

Reply:
I think it's a sane starting point, in the majority of cases the control plane should already have the bootstrap token installed - though I think it might be better to not fail provisioning by failing kubelet startup if we fail to validate, and rather treat this as a best-effort type approach, that way we don't have to be as worried about this and we can also take a data driven approach as to what the real limit should be after having guest agent event telemetry once this is rolled out widely